### ISSUE TITLE
Reading in vectors after configuring with --disable-ifem

### ISSUE THREAD

**Issue by dknez** [libMesh/libmesh#1379]
I never used infinite elements, so I decided to reconfigure and build with --disable-ifem. However, now I'm not able to read vectors in that had been written to disk when I had ifem enabled, e.g. I get this error in : "ERROR: variable  does not exist in this system!" in System::read_header.

Is there a potential fix here, or do I need to continue to use --enable-ifem?

**Comment by roystgnr** [comment]
There's a "--read_ifem_systems" command-line option that tells system_io.C to try to read ifem info in files even if the version string doesn't say there's ifem info there.

IIRC that was for backwards compatibility with files written like a *decade* ago, though; I thought we had everything writing out "with infinite elements" or not in the header file these days so that the read could determine that automatically.

**Comment by dknez** [comment]
The issue I'm running into is actually with the reduced basis code. The read/write vector code in rb_evaluation.C does things in a non-standard way I think (see RBEvaluation::read_in_vectors_from_multiple_files). The reason is that we wanted to avoid extra overhead when reading in multiple Xdr files that correspond to the same system, since that happens a lot in the reduced basis code.

So the issue can be replicated as follows:
- Run reduced_basis_ex1 -online_mode 0 with --enable-ifem
- Run reduced_basis_ex1 -online_mode 1 with --disbale-ifem

I'd be interested in working on a fix for this, if you have an idea of what can be done?

**Comment by roystgnr** [comment]
It looks like you're properly using get_io_version_string() to output at write time whether infinite elements are enabled, but at read time you're *still* using get_io_version_string(), rather than reading what's in the file?  If I'm correct, then changing that should fix things.

**Comment by dknez** [comment]
Yep, you're right! Now that I've reconfigured without ifem, get_io_version_string() omits " with infinite elements" at read time. If I add that back in then it works fine.

As you said, the right solution is to read the header file to determine if we need to include ifem or not. However, when I try to do this, e.g. via:

```
Xdr io ("header.xdr", DECODE);
io.data(version);
```
then version is just empty. Any idea why that would be the case?

If you'd like to try this yourself, I can send you a header.xdr that I tried this on if you like, or you can use the one generated by reduced_basis_ex1 (the file offline_data/bf_header.xdr).

**Comment by roystgnr** [comment]
Could you use xda, to make the debugging a bit easier?  I've no idea why you'd get an empty string.  We don't have anything that gets written to the header files *before* the version string, right?

**Comment by dknez** [comment]
OK, I see the issue. I looked at the header.xda file from reduced_basis_ex1, which looks like this:

```
1	 # No. of Variables in System "RBConvectionDiffusion"
u	 #   Name, Variable No. 0, System "RBConvectionDiffusion"
0	 # vector length
	 #     Subdomains, Variable "u", System "RBConvectionDiffusion"
1	 #     Approximation Order, Variable "u", System "RBConvectionDiffusion"
0	 #     FE Family, Variable "u", System "RBConvectionDiffusion"
0	 # No. of Additional Vectors, System "RBConvectionDiffusion"
```

There is no "version" string in there at all... System::write_header takes a "version" argument but it looks like that is unused (confusing!). Instead I need to add the version to the Xdr before calling write_header.

I'll see if I can make a PR to fix this, though it may be tricky with backwards compatibility.

**Comment by dknez** [comment]
I'll close this. It isn't a general bug, it was just an issue in the reduced basis vector read/write, and should be resolved by PR #1380.
